src/V3HierBlock.cpp

```diff
@@ -138,6 +143,22 @@ V3HierBlock::StrGParams V3HierBlock::stringifyParams(const GParams& gparams, boo
     return strParams;
 }
 
+V3HierBlock::StrGParams
+V3HierBlock::stringifyParams(const V3HierBlockParams::GTypeParams& params) {
+    StrGParams strParams;
+
+    for (const AstParamTypeDType* const gparam : params) {
+        if (const AstBasicDType* const basicp = VN_CAST(gparam->childDTypep(), BasicDType)) {
```

Comment:
Add skipRefp() here - and test a typedef of a basic type.  This is very common, so don't want it unsupported.

Reply:
I'd changed the emitting method to use `V3EmitV::verilogForTree` instead of `prettyDTypeName`. This way, recursive types should work too.